src/isomorphism/vf2.rs

```diff
@@ -41,6 +38,48 @@ use crate::iterators::NodeMap;
 
 type StablePyGraph<Ty> = StableGraph<PyObject, PyObject, Ty>;
 
+/// Returns the adjacency_matrix of a graph with `(i, j)` entry equal
+/// to number of edges from node `i` to node `j`.
+fn adjacency_matrix<Ty: EdgeType>(graph: &StablePyGraph<Ty>) -> Vec<usize> {
```

Comment:
I'm wondering if instead of using a matrix it would be better to fix #367 as part of this too and just query the graph object to get the edge count on demand. Not sure what the performance difference will be though, but it will at least scale better in the amount of memory required.

Reply:
Yeah, I thought about it but in the interest of time, I decided to push this fix and in a follow up, once we also have a serious benchmark, we can check if there is a time performance difference using something like `HashMap`